### PR TITLE
Cleaning up codes

### PR DESCRIPTION
**Pull request by mandurisaimounika**

### CODE REVIEW

**Comment by mandurisaimounika** [comment]
@marcoow the codes has been deleted and re-coded again with optimizations.
The "Sign Up form " and "Footer" codes are pending
Screenshot 2021-06-23 at 10 30 26

**Comment by mandurisaimounika** [comment]
@marcoow all the html and css changes are done. Testing the build.
Screenshot 2021-06-23 at 16 43 06
Screenshot 2021-06-23 at 16 43 35

File: images/2021 _ motif _ decorative element.svg [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (marcoow) [inline images/2021 _ motif _ decorative element.svg:1]: we should make sure there are no spaces in file names.

**Review by marcoow — COMMENTED** [review]
The code looks much better now! I think there are some smaller visual issues still:

* I guess the headers should have some more space around them? Bildschirmfoto 2021-06-24 um 15 21 47
* Some of the sponsor logos seem to be outside the viewport for some screen sizes: Bildschirmfoto 2021-06-24 um 15 20 04

I think in general, we shouldn't be using pixel values in CSS but units like `em` and `rem`. I'm not the most knowledgeable person on the team but I'm sure there'll be someone who can explain and help with this in our Slack in #tech or #ember.

File: images/2021 _ motif _ decorative element.svg (hunk 1 above)
Comment (mandurisaimounika) [reply images/2021 _ motif _ decorative element.svg]: - Headers spacing has been resolved
Screenshot 2021-06-24 at 17 58 22

- The sponsors logo looks fine for me in all the screen viewports. Please mention the screen size in which it is breaking
Screenshot 2021-06-24 at 17 58 03

- I have used "interpolate" in css which will not accept the "em" and "rem". This is the reason I am using the pixel values.
- The white spaces in the file name is also fixed
Screenshot 2021-06-24 at 18 07 49

Comment (marcoow) [reply images/2021 _ motif _ decorative element.svg]: I'm not sure what `interpolate(…` is tbh and can't find documentation on it – could you link to some docs?

Comment (mandurisaimounika) [reply images/2021 _ motif _ decorative element.svg]: https://www.geeksforgeeks.org/sass-interpolation/ and https://sass-lang.com/documentation/interpolation 
These are the documentations related to the "Interpolation" in Sass.

Comment (marcoow) [reply images/2021 _ motif _ decorative element.svg]: Ah, I get it now – there's a mixin called `interpolate`! Can you move that out of `_reset.css`? "reset" in CSS usually refers to resetting all default browser styles to a common baseline and having mixins defined in a file called `_reset.css` is confusing.

I'm also not sure what that mixin does tbh. I think it'd be better to use relative units.

Comment (mandurisaimounika) [reply images/2021 _ motif _ decorative element.svg]: Yes, it is a mixin. I moved out the mixin code from "_reset.css".

File: css/_mixins.scss [hunk 2]
```diff
@@ -0,0 +1,124 @@
+@mixin interpolate($property, $min-screen, $min-value, $max-screen, $max-value, $easing : "linear", $bending-points : 2) {
```
Comment (marcoow) [inline css/_mixins.scss:1]: As discussed, we should be removing this mixin (and the other related ones). It's extremely complex code that also seems to be made for animations which we don't have anyway.

File: css/_mixins.scss [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  @media only screen and (min-width: $tab-breakpoint) {
+    @content;
+  }
+}
+
+@mixin mobile {
+    @media only screen and (min-width: $mobile-breakpoint) {
+      @content;
+    }
+  }
```
Comment (marcoow) [inline css/_mixins.scss:124]: I think these are good to have 👍

File: css/_reset.scss [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+}
+
+svg {
+	overflow : hidden;
+	vertical-align : middle;
+}
+
+html {
+	scroll-behavior : smooth;
+}
```
Comment (marcoow) [inline css/_reset.scss:336]: Only this part is what's commonly referred to as _"reset.css"_ – I think everything else should be removed from this file as it's a bit irritating to have it in here. The classes below could maybe go into `common.scss` and the mixins at the top should likely just be deleted – see comment above.

**Comment by marhigh** [comment]
There are a lot of elements that are not 1:1 with the design, but I would be unable to do a proper design review until Friday which is too late for @mandurisaimounika 's timeline. Therefore, to be practical, my only request would be to fix the primary CTA of the hero in a small viewport as it's currently unreadable: 

Screen Shot 2021-06-29 at 17 53 52

**Review by marhigh — APPROVED** [review]

File: css/_reset.scss (hunk 4 above)
Comment (mandurisaimounika) [reply css/_reset.scss]: @marcoow I removed all the codes related to the "interpolate" mixin.

As per your request, changed the pixel values to rem.

**Comment by mandurisaimounika** [comment]
> There are a lot of elements that are not 1:1 with the design, but I would be unable to do a proper design review until Friday which is too late for @mandurisaimounika 's timeline. Therefore, to be practical, my only request would be to fix the primary CTA of the hero in a small viewport as it's currently unreadable:
> 
> Screen Shot 2021-06-29 at 17 53 52

The primary CTA has been fixed.

**Comment by mandurisaimounika** [comment]
> There are a lot of elements that are not 1:1 with the design, but I would be unable to do a proper design review until Friday which is too late for @mandurisaimounika 's timeline. Therefore, to be practical, my only request would be to fix the primary CTA of the hero in a small viewport as it's currently unreadable:
> 
> Screen Shot 2021-06-29 at 17 53 52

The primary CTA has been fixed.
Screenshot 2021-06-30 at 10 40 29

File: css/_reset.scss [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+	&::after {
+		content : "";
+		clear : both;
+		display : block;
+	}
+}
+
+.hidden {
+	display : none !important;
+}
```
Comment (marcoow) [inline css/_reset.scss:198]: These should also go in the general CSS file since they are not part of reset.

File: css/_variables.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;700&display=swap');
```
Comment (marcoow) [inline css/_variables.scss:1]: I think this should go into the general CSS files since it't not a variable.

**Review by marcoow — COMMENTED** [review]
Please check the visual appearance for all viewport sizes again. Some sponsor logos are outside of the viewport for some sizes, the ticket button changes shape 1 second after loading the page and the venue image should have the same height as the text and map next to it. Unfortunately I don't have a lot of time now to check things but it'd be good if you reviewed everything carefully again and/or asked someone from the team to do it.

File: css/_reset.scss (hunk 5 above)
Comment (mandurisaimounika) [reply css/_reset.scss:198]: Can you please mention the viewport sizes for which the sponsor logos are breaking?

Comment (mandurisaimounika) [reply css/_reset.scss]: I fixed few issues I found while testing.

**Review by marcoow — CHANGES_REQUESTED** [review]
* [x] the header is broken on the content pages:

Bildschirmfoto 2021-08-02 um 10 43 36

* [x] the list on the privacy page is wrong for some screen widths:

Bildschirmfoto 2021-08-02 um 10 43 53

* [x] the map to the conference venue show be as wide a the image on smaller screens:

Bildschirmfoto 2021-08-02 um 10 45 25

* [ ] the logos don't fit the screen on narrow screens:

Bildschirmfoto 2021-08-02 um 10 45 39

**Comment by mandurisaimounika** [comment]
> * [x]  the header is broken on the content pages:
> 
> Bildschirmfoto 2021-08-02 um 10 43 36
> 
> * [x]  the list on the privacy page is wrong for some screen widths:
> 
> Bildschirmfoto 2021-08-02 um 10 43 53
> 
> * [x]  the map to the conference venue show be as wide a the image on smaller screens:
> 
> Bildschirmfoto 2021-08-02 um 10 45 25
> 
> * [ ]  the logos don't fit the screen on narrow screens:
> 
> Bildschirmfoto 2021-08-02 um 10 45 39

@marcoow the changes requested above are fixed except for the sponsors logo. The sponsors logo is working fine for me in all the viewports in Chrome as well as in Safari.

**Comment by marcoow** [comment]
Safari window made as narrow as possible with 2 tabs open:

Bildschirmfoto 2021-08-04 um 10 52 44

**Comment by marcoow** [comment]
hm, seems to be caused by dragging the window…

**Review by marcoow — APPROVED** [review]

**Pull request merged**